src/hooks/useSearch.ts

```diff
@@ -1,26 +1,37 @@
-import { useState, useEffect } from "react";
+import { useState, useEffect, useCallback } from "react";
 import { useDebounce } from "@uidotdev/usehooks";
 
 type UseSearchReturn = {
   filter: Record<string, string> | undefined;
-  handleSearch: (key: string, value: string) => void;
+  handleSearch: (key: string, value: string, debounceTime?: number) => void;
 };
 
 const useSearch = (
   onSearch: (filter: Record<string, string> | undefined) => void,
+  defaultDebounceTime = 500,
 ): UseSearchReturn => {
   const [filter, setFilter] = useState<Record<string, string> | undefined>(
     undefined,
   );
-  const debouncedSearchTerm = useDebounce(filter, 500);
+  const [debounceTime, setDebounceTime] = useState<number>(defaultDebounceTime);
+  const debouncedSearchTerm = useDebounce(filter, debounceTime);
 
   useEffect(() => {
     onSearch(debouncedSearchTerm);
   }, [debouncedSearchTerm, onSearch]);
 
-  const handleSearch = (key: string, value: string) => {
-    setFilter((prevFilter) => ({ ...prevFilter, [key]: value }));
-  };
+  const handleSearch = useCallback(
+    (key: string, value: string, debounceTime?: number) => {
+      if (debounceTime !== undefined) {
+        setDebounceTime(debounceTime);
+      } else {
+        setDebounceTime(defaultDebounceTime);
+      }
```

Comment:
I think it's safe to assume that `debounceTime` will never be undefined. Since you're setting the `defaultDebounceTime` on the initial component render it's bound to be defined all the time.

Reply:
![image](https://github.com/user-attachments/assets/51a870fb-be3b-44d7-befd-3fd22a4728d9)
I was testing it multiple times and this is why `debounceTime` can be `undefined`